pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java

```diff
@@ -622,6 +632,22 @@ public static void increasingQPSQueryRunner(PerfBenchmarkDriverConf conf, String
     }
   }
 
+  private static Stream<String> makeQueries(List<String> inputs, QueryMode queryMode, int queryCount) {
+    queryCount = queryCount > 0 ? queryCount : inputs.size();
+
+    switch (queryMode) {
+      case LIST:
+        return inputs.stream().limit(queryCount);
+
+      case SAMPLE:
+        Random r = new Random(inputs.hashCode()); // anything deterministic will do
```

Comment:
This hashCode could be quite expensive, maybe just use `new Random(0)`?

Reply:
sure. since this is a one-off shouldn't really impact perf at all